models/Asset.php

```diff
@@ -537,18 +536,15 @@ public function save()
                     // if the old path is different from the new path, update all children
                     $updatedChildren = [];
                     if ($oldPath && $oldPath != $this->getRealFullPath()) {
-                        $differentOldPath = $oldPath;
-
                         try {
                             $storage->move($oldPath, $this->getRealFullPath());
+                            $differentOldPath = $oldPath;
```

Comment:
How are all the changes in `models/Asset.php` related to this PR? Maybe a merging issue? This changes were introduced in #9693